packages/components/src/VirtualizedList/utils/resizable.js

```diff
@@ -0,0 +1,267 @@
+ * @param {integer} index
+ * @param {array} columnsWidths
+ */
+const getEnlargeIndexRight = (index, columnsWidths) => {
+	for (let i = index + 1; i < columnsWidths.length; i += 1) {
+		// Enlarge the nearest right column.
+		if (columnsWidths[i].resizable && columnsWidths[i].width > getMinimumWidth(columnsWidths[i])) {
+			return i;
+		}
+		// Enlarge the last column only if trigger by the before last column.
+		if (
+			index === columnsWidths.length - 2 &&
+			i === columnsWidths.length - 1 &&
+			columnsWidths[i].resizable
+		) {
+			return i;
+		}
+	}
+	return -1;
+};
+
+/**
+ * Search for the nearest index shrinkable on the left.
+ * @param {integer} index
+ * @param {array} columnsWidths
+ */
+const getShrinkIndexLeft = (index, columnsWidths) => {
+	for (let i = index; i >= 0; i -= 1) {
+		const previous = i - 1;
+		// Shrink column at left.
+		if (columnsWidths[i].resizable && columnsWidths[i].width > getMinimumWidth(columnsWidths[i])) {
+			return i;
+			// Shrink the after first colum on the left,
+			// if the previous one is already at minimal width.
+		} else if (
+			columnsWidths[i] === getMinimumWidth(columnsWidths[i]) &&
+			previous < 0 &&
+			columnsWidths[previous].resizable &&
+			columnsWidths[previous].width > getMinimumWidth(columnsWidths[previous])
+		) {
+			return previous;
+		}
+	}
+	return -1;
+};
+
+/**
+ * Set the collection with the value, and return it.
+ * @param {boolean} resized the value to set.
+ */
+const setResized = resized => column => ({ ...column, resized });
+
+/**
+ * Set all the array elements to resized false.
+ * @param {array} columnsWidths
+ */
+const resetResized = columnsWidths => columnsWidths.map(setResized(false));
+
+const addWidth = percentDelta => value => percentDelta + value;
+
+const subtractWidth = percentDelta => value => value - percentDelta;
+
+/**
+ * Calculate the new width the parameter function, and assign the result to the given column.
+ * @param {function} calcFn
+ */
+const calculateWidth = calcFn => column => {
+	const { width, minWidth = MINIMUM_COLUMN_WIDTH } = column;
+	if (width >= minWidth) {
+		const calculatedWidth = calcFn(width);
+		return {
+			...column,
+			width: getWidth(calculatedWidth, minWidth),
+		};
+	}
+	return column;
+};
+
+/**
+ * Set a new column to the current list.
+ * @param {array} columnsWidths
+ * @param {integer} index
+ */
+const setColumn = (columnsWidths, index) => column => {
+	// eslint-disable-next-line no-param-reassign
+	columnsWidths[index] = column;
+	return columnsWidths;
+};
+
+/**
+ * Get the nearest column index to shrink on the left side of the given index,
+ * and update the corresponding column with computed width.
+ * @param {number} deltaX
+ * @param {integer} index
+ */
+const shrinkLeftColumn = (deltaX, index) => columnsWidths => {
+	const shrinkIndexLeft = getShrinkIndexLeft(index, columnsWidths);
+	if (shrinkIndexLeft >= 0) {
+		flow([
+			calculateWidth(subtractWidth(Math.abs(deltaX))),
```

Comment:
Everytime this function is executed, deltaX is negative. So `subtractWidth(Math.abs(deltaX))`= `addWidth(deltaX)`